pandas/core/generic.py

```diff
@@ -2146,8 +2147,10 @@ def tail(self, n=5):
         Returns last n rows
         """
         l = len(self)
-        if l == 0 or n == 0:
+        if l == 0:
```

Comment:
I think it was due to the `n == 0 or n == l` check that was there before. Even in that case there's no need to avoid the `iloc` in the latter case; code is simpler without it.